jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -525,3 +526,36 @@ def _live_update(self, event={}):
         for mark in self.marks.values():
             mark.update_xy(sp.spectral_axis.value, sp.flux.value)
             mark.visible = True
+
+    def translate_units(self, spectral_cube, collapsed_spec):
+        if collapsed_spec._unit == u.MJy / u.sr:
+            # MJy to Jy
+            collapsed_spec._data *= 10e6
+
+            # spectral_cube.meta['_pixel_scale_factor'] / 1, where 1 = default scale factor
+            collapsed_spec._data *= spectral_cube.meta['_pixel_scale_factor']
+
+            # per https://jwst-docs.stsci.edu/jwst-near-infrared-camera/nircam-performance/nircam-absolute-flux-calibration-and-zeropoints # noqa
+            # MJy/sr is default, to measure photometry in MJy, convert the image to units
+            # of MJy/pixel by multiplying by the average area of a pixel in sr, a constant
+            # provided in the FITS header keyword PIXAR_SR
+            collapsed_spec._data *= spectral_cube.meta['PIXAR_SR']
+
+            # manually change units
+            collapsed_spec._unit = u.Jy / u.pix
+
+        elif collapsed_spec._unit == u.Jy / u.pix:
+            # Jy to MJy
+            collapsed_spec._data /= 10e6
+
+            # 1 / spectral_cube.meta['_pixel_scale_factor'], where 1 = default scale factor
+            collapsed_spec._data /= spectral_cube.meta['_pixel_scale_factor']
+
+            # refer to above if-statement regarding meta['PIXAR_SR']
+            collapsed_spec._data /= spectral_cube.meta['PIXAR_SR']
+
+            # manually change units
+            collapsed_spec._unit = u.MJy / u.sr
+
+        elif collapsed_spec._unit != u.Jy / u.pix and collapsed_spec._unit != u.MJy / u.sr:
+            raise ValueError(f"Units '{collapsed_spec._unit}' are not MJy/sr nor Jy/pix.")
```

Comment:
can this not be generalized to any flux or surface brightness units?

Reply:
Should be doable and generalizing here is a better idea